src/services/functionAppService.ts

```diff
@@ -48,68 +49,34 @@ export class FunctionAppService extends BaseService {
     await this.sendApiRequest('POST', syncTriggersUrl);
   }
 
-  async cleanUp(functionApp) {
-    this.serverless.cli.log('Cleaning up existing functions');
-    const deleteTasks = [];
-
-    const serviceFunctions = this.serverless.service.getAllFunctions();
-    const deployedFunctions = await this.listFunctions(functionApp);
-
-    deployedFunctions.forEach((func) => {
-      if (serviceFunctions.includes(func.name)) {
-        this.serverless.cli.log(`-> Deleting function '${func.name}'`);
-        deleteTasks.push(this.deleteFunction(func.name));
-      }
-    });
-
-    return await Promise.all(deleteTasks);
-  }
-
-  async listFunctions(functionApp) {
-    const getTokenUrl = `${this.baseUrl}${functionApp.id}/functions?api-version=2016-08-01`;
-    const response = await this.sendApiRequest('GET', getTokenUrl);
-
-    return response.data.value || [];
-  }
-
-  async uploadFunctions(functionApp) {
-    this.serverless.cli.log('Creating azure functions');
-
-    // Perform additional operations per function
-    const serviceFunctions = this.serverless.service.getAllFunctions();
-    const uploadTasks = serviceFunctions.map((functionName) => this.uploadFunction(functionApp, functionName));
-
-    return await Promise.all(uploadTasks);
-  }
-
-  async uploadFunction(functionApp, functionName) {
-    this.serverless.cli.log(`-> Creating function: ${functionName}`);
-
-    const scmDomain = functionApp.enabledHostNames[0];
+  async zipDeploy(functionApp) {
+    const functionAppName = functionApp.name;
+    this.serverless.cli.log(`Deploying zip file to function app: ${functionAppName}`);
 
     // Upload function artifact if it exists, otherwise the full service is handled in 'uploadFunctions' method
-    const functionZipFile = this.serverless.service['functions'][functionName].package.artifact || this.serverless.service['artifact'];
+    const functionZipFile = this.serverless.service['artifact'];
     if (functionZipFile) {
-      this.serverless.cli.log(`-> Uploading function package: ${functionName}`);
+      this.serverless.cli.log(`-> Uploading ${functionZipFile}`);
 
+      const uploadUrl = `https://${functionAppName}${constants.scmDomain}${constants.scmZipDeployApiPath}`;
+      this.serverless.cli.log(`-> Upload url: ${uploadUrl}`);
+
+      // https://github.com/projectkudu/kudu/wiki/Deploying-from-a-zip-file-or-url
       const requestOptions = {
-        method: 'PUT',
-        uri: `https://${scmDomain}/api/zip/site/wwwroot/${functionName}/`,
+        method: 'POST',
+        uri: uploadUrl,
         json: true,
         headers: {
           Authorization: `Bearer ${this.credentials.tokenCache._entries[0].accessToken}`,
-          Accept: '*/*'
+          Accept: '*/*',
+          "Content-type": "application/octet-stream"
```

Comment:
not sure if it matters to Azure, but typically multi-word headers are camel case: `Content-Type`

nitpic: Different than TypeScript, but it looks like the convention is to use single quotes vs. double quotes.